### PR TITLE
remote deploy/destroy improvements

### LINKED ISSUE #3358: Remote deployment - build improvements
I can't run okteto build inside a remote deployment.
I want that my `deploy.dockerignore` to be call `.oktetodeployignore`
I want to set my `deploy.image` from one of the images built in the build step

## Acceptance criteria

- [x] okteto build should be available inside the deploy section
- [x] rename dockerignore for deploy to `.oktetodeployignore`
- [x] `deploy.image` can be an okteto build env var

### PR DESCRIPTION
**Pull request by AdrianPedriza**
# Proposed changes

Fixes #3358

### CODE REVIEW

File: cmd/deploy/remote.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func (rd *remoteDeployCommand) createDockerignoreIfNeeded(cwd, tmpDir string) er
 			return err
 		}
 
-		err = afero.WriteFile(rd.fs, fmt.Sprintf("%s/%s", tmpDir, dockerignoreName), dockerignoreContent, 0600)
+		err = afero.WriteFile(rd.fs, fmt.Sprintf("%s/%s", tmpDir, ".dockerignore"), dockerignoreContent, 0600)
```
Comment (ifbyol) [inline cmd/deploy/remote.go:190]: I think this was the only place where we being using `dockerignoreName` constant. Should we delete that cons?

**Review by jLopezbarb — APPROVED** [review]

File: cmd/deploy/remote.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ WORKDIR /okteto/src
 ENV OKTETO_INVALIDATE_CACHE {{ .RandomInt }}
 RUN okteto deploy --log-output=json {{ .DeployFlags }}
 `
-	dockerignoreName = "deploy.dockerignore"
+	dockerignoreName = ".oktetodeployignore"
```
Comment (ifbyol) [inline cmd/deploy/remote.go:65]: Should we change also the value here? https://github.com/okteto/okteto/blob/adrian/deploy-destroy-same-way/cmd/destroy/remote.go#L52

Comment (AdrianPedriza) [reply cmd/deploy/remote.go]: oh yes, I forgot the change for destroy in remote! Thanks @ifbyol

File: cmd/deploy/remote.go (hunk 1 above)
Comment (AdrianPedriza) [reply cmd/deploy/remote.go:188]: It is used just before this line but I think it is not necessary for now to make it constant if it is only used once

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**